### PR TITLE
Allow GlobalCourseCreatorRole and CourseCreatorRole to set course end date

### PR DESCRIPTION
**Pull request by burhandodhy**
This PR allows Global Course Creator and Course Creator role to set the Course Enrollment End Date.

We only add the `CourseCreatorRole` check as Global Course Creator role also has Course Creator role.

Jira ticket link
https://edlyio.atlassian.net/browse/EDLY-1975

### CODE REVIEW

File: cms/djangoapps/contentstore/views/tests/test_access.py [hunk 1]
```diff
@@ -48,3 +58,20 @@ def test_get_user_role_staff(self):
             'staff',
             get_user_role(self.staff, self.course_key)
         )
+
+    def test_enrollment_end_editable(self):
+        """
+        Verifies if user can edit enrollment end date.
+        """
+        add_users(self.global_admin, CourseCreatorRole(), self.course_creator)
+        assert True == GlobalStaff().has_user(
```
Comment (iam-mhaseeb) [inline cms/djangoapps/contentstore/views/tests/test_access.py:67]:
```
assert  GlobalStaff().has_user(
                self.course_creator
            ) or CourseCreatorRole().has_user(
                self.course_creator
            ) or not marketing_site_enabled
```

File: cms/djangoapps/contentstore/views/tests/test_access.py [hunk 2]
```diff
@@ -48,3 +58,20 @@ def test_get_user_role_staff(self):
...
+                self.course_creator
+            ) or CourseCreatorRole().has_user(
+                self.course_creator
+            ) or not marketing_site_enabled
+        add_users(self.global_admin, GlobalCourseCreatorRole(), self.global_course_creator)
+        assert True ==  GlobalStaff().has_user(
```
Comment (iam-mhaseeb) [inline cms/djangoapps/contentstore/views/tests/test_access.py:73]:
```
assert GlobalStaff().has_user(
                self.global_course_creator
            ) or GlobalCourseCreatorRole().has_user(
                self.global_course_creator
            ) or not marketing_site_enabled
```

**Review by iam-mhaseeb — APPROVED** [review]
Just 2 suggestions otherwise LGTM 👍

File: cms/djangoapps/contentstore/views/tests/test_access.py (hunk 1 above)
Comment (burhandodhy) [reply cms/djangoapps/contentstore/views/tests/test_access.py:67]: done

File: cms/djangoapps/contentstore/views/tests/test_access.py (hunk 2 above)
Comment (burhandodhy) [reply cms/djangoapps/contentstore/views/tests/test_access.py]: done

**Review by asadmanzoor93 — APPROVED** [review]

**Review by rehan99000 — APPROVED** [review]

**Review by zubair-arbi — APPROVED** [review]
LGTM 👍 

@burhandodhy Please rebase and squash commits to one verbose commit and merge it after testing.

**Pull request merged**